python-lib/ontology_tagging_formatting.py

```diff
@@ -11,49 +11,51 @@
 from time import perf_counter
 import logging
 import json
-from plugin_io_utils import move_columns_after, unique_list
+from plugin_io_utils import move_columns_after, unique_list, get_keyword, get_sentence
 from spacy_tokenizer import MultilingualTokenizer
 
+
 class Formatter:
     def __init__(
         self,
         language: AnyStr,
-        splitted_sentences_column: AnyStr,
         tokenizer: MultilingualTokenizer,
-        matcher_dict: dict,
-        keyword_to_tag: dict,
         category_column: AnyStr,
+        case_insensitivity: bool,
+        text_column_tokenized: AnyStr,
+        text_lower_column_tokenized: AnyStr = None,
+        keyword_to_tag: dict = None,
+        matcher_dict: dict = None,
     ):
         store_attr()
         self.output_df = pd.DataFrame()
 
-    def _apply_matcher(
+    def _get_document_language(
         self, row: pd.Series, language_column: AnyStr = None
-    ) -> Tuple[AnyStr, List]:
-        """Apply matcher to the document and returns it
-        Args:
-            row: pandas.Series, document from text_df
-            language_column: if not None, document is matched with the language_column indication
-                             else, document is matched with the formatter language attribute
-        Returns: language of the document and tagged document
-        """
-        language = (
-            row[language_column]
-            if self.language == "language_column"
-            else self.language
-        )
-        document = list(
-            self.tokenizer.spacy_nlp_dict[language].pipe(
-                row[self.splitted_sentences_column]
-            )
+    ) -> AnyStr:
+        """Return the language of the document in the row"""
+        return row[language_column] if language_column else self.language
+
+    def _get_document_to_match(self, row: pd.Series) -> List:
+        """Return the original document (as list of sentences) or, the lowercase one"""
+        if self.case_insensitivity:
+            return row[self.text_lower_column_tokenized]
+        else:
+            return row[self.text_column_tokenized]
+
+    def _columns_to_drop(self) -> None:
```

Comment:
Nitpick: better to use an action verb for all function / method names
```suggestion
    def _get_columns_to_drop(self) -> None:
```